### DISCUSSION TITLE
How can I check if a req has a body without trying to parse it in the first place?

### DISCUSSION THREAD

**Discussion by jjoselv** [mswjs/msw#1348] [Q&A] [answered]
Do you know if something like `hasBody()` exists?

**Comment by kettanaito** [comment, marked as answer]
Hi, @jjoselv. 

When handling requests on the web the only way you can check for a body without reading it is by reading the `Content-Length` header sent in the response. All the other APIs require you to read the body. It makes sense, as body may be represented as a stream, and you have to read that stream to know if there's anything there at all. 

Strictly speaking, this is not an MSW-related question. Is there a particular use case you're trying to achieve with the library? If so, please share it, alongside your attempts to do so (code/reproduction repo).

**Reply by jjoselv** [reply]
Fair enough ☺️ that's for explaining!

**Reply by kettanaito** [reply]
In real world, you can make a `HEAD /resource` request that will return you response headers without initiating the body reading. That's where you check `Content-Length`, if necessary.

**Reply by camillecroci** [reply]
I'd be careful with the `content-length` . I tried to use it and ended up with a bug because I would get a response without the property: 
https://stackoverflow.com/questions/11375507/getting-response-of-http-request-without-content-length

**Reply by kettanaito** [reply]
You can also check on `request.body`. If it's `ReadableStream`, the request has body. If it's `null`, it doesn't.
